### PR TITLE
New notifier: Google Chat

### PR DESCRIPTION
**Pull request by glasnt**
Adds a new notifier type for Google Chat messages. 

image

Pending action items (would appreciate any and all help with these!)

 - [ ] unit tests
 - [ ] github checks
 - [ ] checking payload conversion googlechat/main.go:81
 - [ ] (possibly a feature for later) functionality to get trigger information (requires https://pkg.go.dev/google.golang.org/api/cloudbuild/v1)
 - [ ] documentation (internal)

### CODE REVIEW

File: googlechat/main.go [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+	}
+
+	log.Infof("sending Google Chat webhook for Build %q (status: %q)", build.Id, build.Status)
+	msg, err := g.writeMessage(build)
+	if err != nil {
+		return fmt.Errorf("failed to write Google Chat message: %w", err)
+	}
+	//TODO(glasnt) unsure if this is best practice.
+	payload := new(bytes.Buffer)
+	json.NewEncoder(payload).Encode(msg)
```
Comment (prabenzom) [inline googlechat/main.go:83]: I believe encode returns and error and you should check for that case?

File: googlechat/main_test.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
+package main
+
+import (
+	"testing"
+)
+
+func TestWriteMessage(t *testing.T) {
+	/*
```
Comment (prabenzom) [inline googlechat/main_test.go:8]: hmm, why are these commented out?

**Review by prabenzom — CHANGES_REQUESTED** [review]
Generally lgtm, just some nits. Were you planning on writing unit tests for this?

**Comment by glasnt** [comment]
Thanks for the review @prabenzom! I'll comment inline for each item, but a general note: I'm not a golang developer so I had to upskill a lot just to get this PR together. Happy to learn more to help get this through, and would appreciate the assistance!

File: googlechat/main_test.go (hunk 2 above)
Comment (glasnt) [reply googlechat/main_test.go:8]: These were originally from the Slack version, just bitrot. I'll work at adding unit tests

File: googlechat/main.go (hunk 1 above)
Comment (glasnt) [reply googlechat/main.go:83]: Will do!

File: googlechat/main.go [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+	if !g.filter.Apply(ctx, build) {
+		return nil
+	}
+
+	log.Infof("sending Google Chat webhook for Build %q (status: %q)", build.Id, build.Status)
+	msg, err := g.writeMessage(build)
+	if err != nil {
+		return fmt.Errorf("failed to write Google Chat message: %w", err)
+	}
+	//TODO(glasnt) unsure if this is best practice.
```
[the remark this answers is not in the document]
Comment (glasnt) [reply googlechat/main.go:81]: I wasn't sure if there was a better way to construct this object, but based on your later comment it's okay?

**Review by prabenzom — APPROVED** [review]
Looks great, thank you for doing this!

**Pull request merged**